### DISCUSSION TITLE
With the recent changes, is redirecting port 443 enough?

### DISCUSSION THREAD

**Discussion by top-gun** [llamafilm/tesla-http-proxy-addon#60] [General]
I run my HA-server behind a Fritz!Box DSL router, connected via dualstack to the internet. For the initial setup of the Fleet API token, I configured the HA server as an exposed host, so I didn't run into issues due to missing ports.

With the latest iteration and the need for port 8099 gone, should it be enough just to use port forwarding of port 443? And do I actually need to keep 443 open for token renewal, or can I just close all ports once the API is working?

**Comment by llamafilm** [comment]
Port 8099 is no longer required. 
Port 443 is still required as the car will check it every time you send a command.

**Reply by darek-margas** [reply]
I see you looked into how it works further than me.
Please correct me if I am wrong. I spent a while to install your container alone as I don't use add-ons. Following your manual (but not your bash script), I have got it to the point when my key is in the car and Tesla proxy runs happily. I did not switch my HA to use it, however, so that I did not test if that actually does the job.
Now, I understood redirection to 8099 was to download key and to enroll to Tesla API only and as such may remain available for file just won't install key while flask does not run. That's fine.
I also understood that 443 is required as internal only, for HA to be able to contact and send requests to, hence selfsigned cert used and loaded to HA too. 
Yet you said 443 is required as car will contact it before executing command. I thought it is one way: HA -> tesla-proxy -> car and if there is any bidirectional communication, it will stay wrapped in already established TCP/SSL. 
Am I wrong and tesla proxy has to be available OUTSIDE for the car to initiate connection to? Or I am right and in fact I can even shutdown the proxy hosting key too?

**Reply by llamafilm** [reply]
There are 2 different things running on 443. The http proxy running inside this container will be accessed only by Home Assistant. That’s why the self signed cert is ok.
The public key will be accessed by the vehicle. In my guide, that part is served by nginx. The key file is created by this addon but the addon is not involved in hosting it. So if you’re not using addons, you’ll need to copy that file somewhere and host it with your web server.

**Comment by darek-margas** [comment]
> Port 443 is still required as the car will check it every time you send a command.



> The public key will be accessed by the vehicle. In my guide, that part is served by nginx. The key file is created by this addon but the addon is not involved in hosting it. So if you’re not using addons, you’ll need to copy that file somewhere and host it with your web server.

Thanks, I am pass this point, I used nginx which I use anyway to add ssl to things such as esphome UI. Does public key has to be served after car downloaded it? Or I can disable this file service service until next car to add?

**Reply by llamafilm** [reply]
As I said, the car will check it every time you send a command.
